src/components/ProxyAllowanceCheck.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+import {
+  Text,
+  Card,
+  Button,
+  Grid,
+  Tooltip
+} from '@makerdao/ui-components-core';
+
+import lang from 'languages';
+import useBlockHeight from 'hooks/useBlockHeight';
+import useProxy from 'hooks/useProxy';
+import useTokenAllowance from 'hooks/useTokenAllowance';
+
+import { ReactComponent as Checkmark } from 'images/checkmark.svg';
+import TooltipContents from 'components/TooltipContents';
+
+const SuccessButton = () => {
+  return (
+    <Button variant="primary-outline" width="13.0rem" mt="xs" disabled>
+      <Checkmark />
+    </Button>
+  );
+};
+
+const ProxyAllowanceCheck = ({ symbol, labels, dispatch }) => {
+  const blockHeight = useBlockHeight(0);
+  const { setup_text } = labels;
+
+  const {
+    proxyAddress,
+    setupProxy,
+    proxyLoading,
+    startingBlockHeight,
+    proxyDeployed,
+    proxyErrors
+  } = useProxy();
+
+  const {
+    hasAllowance,
+    setAllowance,
+    allowanceLoading: isSettingAllowance
+  } = useTokenAllowance(symbol);
+
+  async function deployProxy() {
+    await setupProxy();
+    dispatch({
+      type: 'set-proxy-address',
+      payload: { address: proxyAddress }
+    });
+  }
+
+  console.log(
+    '%%2PALuseProxy vals: proxyAddress,proxyLoading,startingBlockHeight,proxyDeployed, proxyErrors',
+    proxyAddress,
+    proxyLoading,
+    startingBlockHeight,
+    proxyDeployed,
+    proxyErrors
+  );
```

Comment:
In this component, the state variables are being updated correctly. I'm wondering if it's because I'm explicitly calling `setupProxy` on line 46.